indico/modules/events/contributions/fields.py

```diff
@@ -52,6 +52,8 @@ def _get_person_link(self, data):
         return super(ContributionPersonLinkListField, self)._get_person_link(data, extra_data)
 
     def _serialize_person_link(self, principal, extra_data=None):
+        if principal.id is None:
+            return super(ContributionPersonLinkListField, self)._serialize_principal(principal.user)
```

Comment:
what happens if there is no user? i.e. if you "enter" someone and then submit the form with validation failing in another field?